ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -855,8 +855,12 @@ mbfl_strpos(
 		needle_u8 = needle;
 	}
 
-	if (needle_u8->len < 1) {
-		result = (size_t) -8;
+	if (needle_u8->len == 0) {
+		if (reverse) {
+			result = haystack_u8->len;
+		} else {
+			result = (size_t) 0;
+		}
```

Comment:
This doesn't take the `offset` parameter into account.

Reply:
Indeed, thanks for noticing, fixed that and amended the commit.